libamqpprox/amqpprox_limitcontrolcommand.cpp

```diff
@@ -29,189 +32,201 @@ namespace amqpprox {
 
 namespace {
 
-void enableConnectionRateLimitBasedOnInput(
-    bool                                                 alarmOnly,
+void handleConnectionLimitAlarm(
     std::istringstream                                  &iss,
     ControlCommandOutput<ControlCommand::OutputFunctor> &output,
-    ConnectionLimiterManager *connectionLimiterManager)
+    ConnectionLimiterManager *connectionLimiterManager,
+    bool                      isDefault,
+    const std::string        &vhostName,
+    bool                      isDisable)
 {
-    std::string subcommand;
-    if (iss >> subcommand) {
-        boost::to_upper(subcommand);
-        if (subcommand == "VHOST") {
-            std::string vhostName;
-            if (!(iss >> vhostName)) {
-                output << "No vhostName specified.\n";
-                return;
-            }
-
-            uint32_t numberOfConnections;
-            if (!(iss >> numberOfConnections && numberOfConnections >= 0 &&
-                  numberOfConnections <
-                      std::numeric_limits<uint32_t>::max())) {
-                output << "Invalid numberOfConnections provided.\n";
-                return;
-            }
-
-            if (alarmOnly) {
-                output << "For vhost " << vhostName << ", "
-                       << connectionLimiterManager
-                              ->addAlarmOnlyConnectionRateLimiter(
-                                  vhostName, numberOfConnections)
-                              ->toString()
-                       << " in alarm only mode.\n";
-                output << "The limiter will only log at warning level with "
-                          "AMQPPROX_CONNECTION_LIMIT as a substring and the "
-                          "relevant limit details, when the new incoming "
-                          "connection violates the specified limit.\n";
-            }
-            else {
-                output << "For vhost " << vhostName << ", "
-                       << connectionLimiterManager
-                              ->addConnectionRateLimiter(vhostName,
-                                                         numberOfConnections)
-                              ->toString()
-                       << "\n";
-            }
-        }
-        else if (subcommand == "DEFAULT") {
-            uint32_t numberOfConnections;
-            if (!(iss >> numberOfConnections && numberOfConnections >= 0 &&
-                  numberOfConnections <
-                      std::numeric_limits<uint32_t>::max())) {
```

Comment:
I did change this logic - since `numberOfConnections` is `uint32_t` it can't be less than 0 or greater than max uint32_t